File: src/config/localConfig.ts
/** Responsibility: read and write local uncommitted config with environment-aware paths. */
import { chmod, mkdir, readFile, writeFile } from "node:fs/promises";
import path from "node:path";

export interface LocalConfig {
  cookie?: string;
  uid?: string;
  defaultLimit?: number;
  loginUrl?: string;
  updatedAt?: string;
}

const DEFAULT_DATA_DIR = path.resolve(process.cwd(), ".local");

export function getLocalDataDir(): string {
  const configuredDir = process.env.WEIBO_CLI_DATA_DIR?.trim();

  return configuredDir ? path.resolve(configuredDir) : DEFAULT_DATA_DIR;
}

export function getLocalConfigPath(): string {
  return path.join(getLocalDataDir(), "weibo-session.json");
}

export async function readLocalConfig(): Promise<LocalConfig | null> {
  try {
    const raw = await readFile(getLocalConfigPath(), "utf8");
    return JSON.parse(raw) as LocalConfig;
  } catch (error) {
    if (isMissingFileError(error)) {
      return null;
    }

    throw error;
  }
}

export async function writeLocalConfig(config: LocalConfig): Promise<string> {
  const dataDir = getLocalDataDir();
  const targetPath = getLocalConfigPath();

  await mkdir(dataDir, { recursive: true });
  await writeFile(targetPath, `${JSON.stringify(config, null, 2)}\n`, {
    encoding: "utf8",
    mode: 0o600
  });

  try {
    await chmod(targetPath, 0o600);
  } catch {
    // Ignore chmod failures on platforms or filesystems that do not support POSIX permissions.
  }

  return targetPath;
}

function isMissingFileError(error: unknown): error is NodeJS.ErrnoException {
  return (
    typeof error === "object" &&
    error !== null &&
    "code" in error &&
    error.code === "ENOENT"
  );
}

File: skills/weibo-cli/scripts/weibo_cli/local_config.py
"""本地配置读写。"""

from __future__ import annotations

import json
import os
from pathlib import Path
from typing import Any


def get_repo_root() -> Path:
    return Path(__file__).resolve().parents[4]


def get_local_data_dir() -> Path:
    configured_dir = os.environ.get("WEIBO_CLI_DATA_DIR", "").strip()
    if configured_dir:
        return Path(configured_dir).expanduser().resolve()
    return get_repo_root() / ".local"


def get_local_config_path() -> Path:
    return get_local_data_dir() / "weibo-session.json"


def read_local_config() -> dict[str, Any] | None:
    path = get_local_config_path()
    if not path.exists():
        return None
    return json.loads(path.read_text(encoding="utf-8"))


def write_local_config(config: dict[str, Any]) -> Path:
    path = get_local_config_path()
    path.parent.mkdir(parents=True, exist_ok=True)
    path.write_text(f"{json.dumps(config, ensure_ascii=False, indent=2)}\n", encoding="utf-8")
    try:
        path.chmod(0o600)
    except OSError:
        pass
    return path